catalyst-gateway/bin/src/db/event/config/key.rs

```diff
@@ -0,0 +1,90 @@
+//! Configuration Key
+
+use std::net::IpAddr;
+
+use serde_json::Value;
+
+/// Configuration key
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) enum ConfigKey {
+    /// Frontend general configuration
+    Frontend,
+    /// Frontend configuration for a specific IP address
+    FrontendForIp(IpAddr),
+}
+
+/// Frontend JSON schema
+const FRONTEND_JSON_SCHEMA: &str = include_str!("jsonschema/frontend.json");
+/// Default frontend configuration
+const FRONTEND_DEFAULT: &str = include_str!("default/frontend.json");
+
+impl ConfigKey {
+    /// Create a `ConfigKey` from ids.
+    pub(crate) fn from_id(id1: &str, id2: &str, id3: &str) -> Option<Self> {
+        match (id1, id2, id3) {
+            ("frontend", "", "") => Some(ConfigKey::Frontend),
+            ("frontend", "ip", ip) => ip.parse().ok().map(ConfigKey::FrontendForIp),
+            _ => None,
+        }
+    }
+
+    /// Convert a `ConfigKey` to its ids.
+    pub(super) fn to_id(&self) -> (String, String, String) {
+        match self {
+            ConfigKey::Frontend => ("frontend".to_string(), String::new(), String::new()),
+            ConfigKey::FrontendForIp(ip) => {
+                ("frontend".to_string(), "ip".to_string(), ip.to_string())
+            },
+        }
+    }
+
+    /// Validate the provided value against the JSON schema.
+    pub(super) fn validate(&self, value: &Value) -> anyhow::Result<()> {
+        let schema: Value = serde_json::from_str(FRONTEND_JSON_SCHEMA)
```

Comment:
This should be driven a match on the enum value of Self.  As there is only one option at the moment, you can use a clippy override to silence the warning.